vmdb/app/models/miq_queue.rb

```diff
@@ -220,10 +220,9 @@ def self.peek(options = {})
       conditions[:priority] || MIN_PRIORITY,
     ]
 
-    args = { :conditions => cond, :order => "priority, id" }
-    args[:select] = select unless select.nil?
-    args[:limit]  = limit || 1
-    MiqQueue.all(args)
+    result = MiqQueue.where(cond).order(:priority, :id).limit(limit || 1)
+    result.select(select) unless select.nil?
```

Comment:
You need to set it back to result else it's a noop.   `result = result.select...`